### PR TITLE
Updating caching documentation, rewriting bits, adding logs image.

### PR DESCRIPTION
**Pull request by ethangj**
Rewriting for clarity, changing order a bit based on common questions asked, adding image for log output.

### CODE REVIEW

File: _posts/docker/tutorials/2015-09-07-caching.md [hunk 1]
```diff
@@ -18,11 +18,11 @@ To follow this tutorial on your own computer, please [install the `jet` CLI loca
 
 ## Using Caching
 
-Please see the [example in the codeship-tool examples repository](https://github.com/codeship/codeship-tool-examples/tree/master/17.caching) of how to use caching.
+As a way to speed up your build pipelines, Codeship  supports remote caching. This means we'll push your final, built image out to your image repository with a special tag: `codeship-cache-$BRANCH`. When using caching, the image push is automatic - there is no need to add a separate push step later in your pipeline.
```
Comment (BoVice) [inline _posts/docker/tutorials/2015-09-07-caching.md]: You have an extra space between "Codeship Supports"

File: _posts/docker/tutorials/2015-09-07-caching.md [hunk 2]
```diff
@@ -18,11 +18,11 @@ To follow this tutorial on your own computer, please [install the `jet` CLI loca
...
 
-As part of your build pipeline, Codeship now supports storing the Docker image cache and retreiving this in order to speed up your builds. To achieve this, the cached builds are pushed to your registry under a caching tag, unique for the specific branch. Before those cached builds are run, the caching tag is pulled which populates the local docker image cache. Depending on the Dockerfile being used, this can speed up the build considerably.
+For future builds, we'll compare the cached image with your new build's Dockerfile to determine  whether there were any changes that invalidate the cache. Once we encounter a change, we will rebuild every remaining layer following the detected change in your Dockerfile. [Read more about optimizing your Dockerfile for caching.]({{ site.baseurl }}/docker/caching/#optimizing-your-build-to-use-the-docker-image-cache)
```
Comment (BoVice) [inline _posts/docker/tutorials/2015-09-07-caching.md]: You have two spaces between "determine whether"

**Comment by BoVice** [comment]
Made two comments on really minor stuff. Besides that GTM. Awesome stuff.

File: .bundle/config [hunk 3]
```diff
@@ -0,0 +1 @@
+--- {}
```
Comment (mlocher) [inline .bundle/config]: Why do we need that file? Was it auto-generated by running `bundle install` (if so we should add it to `.gitignore`), if not, please delete the file again and rebase the branch to not push it to GitHub.

Comment (mlocher) [inline .bundle/config]: Reading through http://bundler.io/v1.3/man/bundle-config.1.html, I'd say we should delete the file, rebase the commits to not include it at all and force push to this branch to have a clean history.

**Comment by ethangj** [comment]
@mlocher is this right or should i also remove the removing bundle config commit and leave no trace?

**Comment by mlocher** [comment]
@ethangj I think you squashed the wrong commits. I cleaned up the branch but had to re-create the commit with your name (that's why you show up as the author, but I'm the committer). I didn't change the article in any way.

 I also noticed that you don't have the correct git user configuration set on your new machine. Please follow https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup#Your-Identity

:shipit:

**Pull request merged**